strawberryfields/compilers/xcov.py

```diff
@@ -92,8 +92,18 @@ def compile(self, seq, registers):
             # determine the modes that are acted on by the symplectic transformation
             used_modes = [x.ind for x in seq[0].reg]
 
+            # Since this compiler does not allow for displacements
+            # when its parameter are passed to the GaussianUnitary compiler
+            # the latter either returns a GaussianTransform + MeasureFock
+            # or just MeasureFock. This is because the GaussianUnitary checks
+            # if the symplectic matrix is just the identity it simply elides it
+
+
             # extract the compiled symplectic matrix
-            S = seq[0].op.p[0]
+            try:
```

Comment:
A much better way to do this would be to check that `seq[0].op` is a `MeasureFock` command, but I am not sure how to do this.

Reply:
`isinstance` is my friend :)